packages/swing-store/src/repair-metadata.js

```diff
@@ -0,0 +1,82 @@
+import { Fail, q } from '@agoric/assert';
+import { makeSwingStore } from './swingStore.js';
+import { assertComplete } from './importer.js';
+
+/**
+ * Given a pre-existing swingstore and a SwingStoreExporter, read in
+ * all the metadata from the exporter and use it to regenerate any
+ * missing metadata records. This can be used to fix the damage caused
+ * by #8025.
+ *
+ * The repair method will call `exporter.getExportData` and examine
+ * all entries to do one of three things:
+ *
+ * 1: kvStore records are ignored (not metadata)
+ * 2: bundle/snapshot/transcript records whose keys already exist will
+ *    be compared against the existing data, and an error thrown if
+ *    they do not match
```

Comment:
Thinking more about this, can we change this behavior slightly to facilitate handling of handover inconsistency:

if the record is for "current" / "in-use" it's ignored (or at most check that hashes match in the case of bundle & snapshots, but not transcripts)

idempotent add (aka check same if present) only for non current / in-use.

That allows to re-run the repair step even if the swing-store committed at end of upgrade block, but somehow the cosmos DB didn't and crashed before.

Reply:
ah, hm, lemme think about this

Reply:
After further discussion, let's not do this.